File: src/main/java/com/nickcoblentz/montoya/settings/ListStringExtensionSetting.java
package com.nickcoblentz.montoya.settings;

import burp.api.montoya.MontoyaApi;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class ListStringExtensionSetting extends GenericExtensionSetting<List<String>> {


    public ListStringExtensionSetting(MontoyaApi api, String name, String storageKey, List<String> defaultValue, ExtensionSettingSaveLocation defaultSaveLocation) {
        super(api, name, storageKey, defaultValue, defaultSaveLocation);
    }

    @Override
    public void loadFromProject() {
        setCurrentValue(parseStringAsList(getApi().persistence().extensionData().getString(getStorageKey())));
    }

    @Override
    public void loadFromPreferences() {
        setCurrentValue(parseStringAsList(getApi().persistence().preferences().getString(getStorageKey())));
    }

    @Override
    public void saveToProject() {
        getApi().persistence().extensionData().setString(getStorageKey(),joinListAsString(getCurrentValue()));
    }

    @Override
    public void saveToPreferences() {
        getApi().persistence().preferences().setString(getStorageKey(),joinListAsString(getCurrentValue()));
    }

    public List<String> parseStringAsList(String values)
    {
        if(values!=null && !values.isEmpty())
            return Arrays.asList(values.split("\n"));
        return new ArrayList<String>();
    }

    public String joinListAsString(List<String> values)
    {
        if(values!=null)
            return String.join("\n",values);
        return "";
    }

    @Override
    public void setCurrentValue(String value) {
        setCurrentValue(parseStringAsList(value));
    }

    @Override
    public List<String> parseString(String value) {
        return parseStringAsList(value);
    }

    public String GetCurrentValueAsString()
    {
        return joinListAsString(getCurrentValue());
    }
}

File: src/main/java/com/nickcoblentz/montoya/settings/ListStringExtensionSetting.kt
package com.nickcoblentz.montoya.settings

import burp.api.montoya.MontoyaApi

class ListStringExtensionSetting(
    api: MontoyaApi,
    name: String,
    storageKey: String,
    defaultValue: List<String>,
    defaultSaveLocation: ExtensionSettingSaveLocation
) :
    GenericExtensionSetting<List<String>>(
        api,
        name, storageKey, defaultValue, defaultSaveLocation
    ) {
    override fun loadFromProject() {
        currentValue = parseStringAsList(api.persistence().extensionData().getString(storageKey))
    }

    override fun loadFromPreferences() {
        currentValue = parseStringAsList(api.persistence().preferences().getString(storageKey))
    }

    override fun saveToProject() {
        api.persistence().extensionData().setString(storageKey, joinListAsString(currentValue))
    }

    override fun saveToPreferences() {
        api.persistence().preferences().setString(storageKey, joinListAsString(currentValue))
    }

    fun parseStringAsList(values: String): List<String> {
        if (values.isNotEmpty()) {
            return values.split("\n")
        }
        return emptyList()
    }

    fun joinListAsString(values: List<String>): String {
        return java.lang.String.join("\n", values)
    }

    override fun setCurrentValue(value: String?) {
        currentValue = if(value!=null) parseStringAsList(value) else currentValue
    }

    override fun parseString(value: String): List<String> {
        return parseStringAsList(value)
    }

    fun getCurrentValueAsString(): String {
        return joinListAsString(currentValue)
    }
}